moped-editor/src/views/projects/projectView/ProjectComponents/utils/form.js

```diff
@@ -55,6 +55,40 @@ export const useComponentOptions = (data) =>
     return options;
   }, [data]);
 
+/**
+ * Take options returned by useComponentOptions and filter them by line representation of the component
+ * currently being edited to keep from switching between lines and points after creation.
+ * @param {Boolean} shouldFilterOptions Should the options be filtered at all
+ * @param {Array} options Component options returned from useComponentOptions for Autocomplete
+ * @param {Boolean} isLineRepresentation is component being edited line represented or not (line or point)
+ * @returns {Array} The options with value, label, and full data object to produce the subcomponents options
+ */
+export const useComponentOptionsFilteredByLineRepresentation = ({
+  shouldFilterOptions,
+  options,
+  isLineRepresentation,
+}) =>
+  useMemo(() => {
+    if (!shouldFilterOptions) return options;
+
+    return options.filter(
+      (component) => component.data.line_representation === isLineRepresentation
+    );
+  }, [shouldFilterOptions, options, isLineRepresentation]);
+
+/**
+ * Take options returned by useComponentOptions and filters out signal components.
+ * @param {Array} options Component Autocomplete options
```

Comment:
Just realized that I've been leaning on VSCode docstring syntax highlighting here and leaving out dashes after the parameter name. 
`* @param {Array} options Component Autocomplete options`
vs.
`* @param {Array} options - Component Autocomplete options`

If this bother you, just say the word, and I'll update this.